### PR TITLE
Bug 1884390 - Adjust Unsupported Error in Translations

### PR DESCRIPTION
**Pull request by ohall-m**
This patch:
* Adds an option in `LocaleUtils` to fetch language names for a given locale.
* Adjusts Translations Unsupported Error to use this option.

### Pull Request checklist

- [X] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1884390

### CODE REVIEW

**Comment by ohall-m** [comment]
Before PR:

After PR (App Language)

(English system language)

After PR:
After PR (App Language)

(German app language on English system language)

After PR  (System Language)

(English system language)

**Review by MatthewTighe — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBinding.kt [hunk 1]
```diff
@@ -133,10 +134,15 @@ class TranslationsDialogBinding(
...
-                    }
+                    sessionTranslationsState.translationEngineState
+                        ?.detectedLanguages?.documentLangTag?.let { docLangTag ->
+                            val documentLocale = Locale.forLanguageTag(docLangTag)
+                            val userLocale = state.browserState.locale ?: LocaleManager.getSystemDefault()
+                            documentLangDisplayName = LocaleUtils.getLocalizedDisplayName(
+                                userLocale = userLocale,
+                                languageLocale = documentLocale,
+                            )
+                        }
```
[the remark this answers is not in the document]
Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBinding.kt]: Updated, ty!

File: fenix/app/src/main/java/org/mozilla/fenix/utils/LocaleUtils.kt [hunk 2]
```diff
@@ -13,6 +13,12 @@ object LocaleUtils {
 
     /**
      * Returns a name for the locale that is appropriate for display to the user.
+     *
+     * For example, in an any language locale, the German language locale or "de" would be
+     * presented as "Deutsch".
```
[the remark this answers is not in the document]
Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/utils/LocaleUtils.kt:19]: Yup, that is correct. I added an additional line: `The display name will be in the language of the specified locale.` to try and help clarify. (We probably should find a better name for that [array](https://searchfox.org/mozilla-mobile/rev/5f71a2ffc106566a706f945437af2212012379f5/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/utils/LocaleUtils.kt#26) based on [these docs as well](https://developers.google.com/style/inclusive-documentation#features-and-users), but trying to keep this patch in scope and a bit timid about adjusting too much L10N code.)

File: fenix/app/src/main/java/org/mozilla/fenix/utils/LocaleUtils.kt [hunk 3]
```diff
@@ -23,6 +29,26 @@ object LocaleUtils {
...
+     * @param languageLocale The locale to localize the language name into the user's language.
+     *
+     * @return A default language name in the specified user's locale language.
+     */
+    fun getLocalizedDisplayName(userLocale: Locale, languageLocale: Locale): String {
+        val displayName = Locale.forLanguageTag(languageLocale.toLanguageTag()).getDisplayName(userLocale)
+        if (userLocale == Locale.ENGLISH) {
+            return LOCALE_TO_DISPLAY_ENGLISH_NAME_MAP[languageLocale.toString()] ?: displayName
+        }
+        return displayName
```
[the remark this answers is not in the document]
Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/utils/LocaleUtils.kt]: Updated, ty!

**Pull request merged**